### DISCUSSION TITLE
Удаление таблиц из базы данных при деинсталяции компонента

### DISCUSSION THREAD

**Discussion by biz87** [Ibochkarev/miniShop2#505] [General]
minishop2 не удаляет таблицы из базы данных при деинсталяции и удалении пакета. 
Предлагаю рассмотреть вопрос о настройке нормального удаления таблиц. 

Но Есть мнение что это и не нужно. 

> вроде как все компоненты так пашут
> пакет тусуется, база остается, если случайно тусанул

**Comment by OlegShchavelev** [comment]
А в чем польза. Человек психанул нажал кнопку удалить и упс. Сейчас он может установить и все восстановится.

**Comment by Ruslan-Aleev** [comment]
В MODX есть несколько типов удаления, сколько лет пользуюсь, до сих пор логику их не понял.
Записи в базе остаются при любом типе удаления? Если да, то как-то странно, логично при удалении сносить все, а при деинсталляции только пакет в админке, оставив записи в базе.

**Reply by Electrica** [reply]
> В MODX есть несколько типов удаления, сколько лет пользуюсь, до сих пор логику их не понял.
> Записи в базе остаются при любом типе удаления? Если да, то как-то странно, логично при удалении сносить все, а при деинсталляции только пакет в админке, оставив записи в базе.

а там как опишешь в ресолвере вроде. типо того https://github.com/Ibochkarev/miniShop2/blob/master/_build/resolvers/resolve.tables.php#L88

но у всех он кажись пустой.

**Comment by CrazyBoy49z** [comment]
Лучше сделать возможность пользователю, иногда бывали случаи когда нужно было переустановить minishop2, тобто полное удаление и установка помогала. Переустановить не работало. В таком случае пользователь без бекапа будет не в шоке куда данные пропали
